### PR TITLE
[Tests-Only] Convert unauthorized userTest to jest

### PR DESCRIPTION
**Pull request by swoichha**
move unauthorized userTests to jest

part of #645 & #646

### CODE REVIEW

File: tests/unauthorized/userTest.js [hunk 1]
```diff
@@ -1,287 +1,365 @@
...
+  const {
+    unauthorizedXmlResponseBody,
+    createProvider
+  } = require('../pactHelper.js')
 
-  const request = function (requestName, method, path) {
-    return {
-      uponReceiving: requestName,
-      withRequest: {
+  const request = async (provider, requestName, method, path, query) => {
```
Comment (dpakach) [inline tests/unauthorized/userTest.js:18]:
```suggestion
  const request = (provider, requestName, method, path, query) => {
```

File: tests/unauthorized/userTest.js [hunk 2]
```diff
@@ -1,287 +1,365 @@
...
+          authorization: invalidAuthHeader
+        },
+        query: query
+      })
+      .willRespondWith({
         status: 401,
         headers: {
-          'Content-Type': 'text/xml; charset=utf-8',
-          'Access-Control-Allow-Origin': origin
+          'Content-Type': 'text/xml; charset=utf-8'
```
Comment (dpakach) [inline tests/unauthorized/userTest.js:32]: import this application/xml header from `pactHelper`

File: tests/unauthorized/userTest.js [hunk 3]
```diff
@@ -1,287 +1,365 @@
...
-    }
+      })
   }
 
-  const usersEndpointPath = Pact.Matchers.term({
-    matcher: '.*\\/ocs\\/v1\\.php\\/cloud\\/users$',
-    generate: '/ocs/v1.php/cloud/users'
-  })
+  const adminUserEndpointPath = MatchersV3.regex(
+    '.*\\/ocs\\/v1\\.php\\/cloud\\/users\\/admin$',
```
Comment (dpakach) [inline tests/unauthorized/userTest.js:39]: seems like the endpoint is always same. why use regex here?

File: tests/unauthorized/userTest.js [hunk 4]
```diff
@@ -1,287 +1,365 @@
...
+    '/ocs/v1.php/cloud/users/admin'
+  )
 
-  const groupsEndpointPath = Pact.Matchers.term({
-    matcher: '.*\\/ocs\\/v1\\.php\\/cloud\\/users\\/' + config.testUser + '\\/groups$',
-    generate: '/ocs/v1.php/cloud/users/' + config.testUser + '/groups'
-  })
+  const usersEndpointPath = MatchersV3.regex(
```
Comment (dpakach) [inline tests/unauthorized/userTest.js:43]: same here I dont't think this requires regex

Comment (dpakach) [reply tests/unauthorized/userTest.js:43]: also do same for other similar examples

File: tests/unauthorized/userTest.js (hunk 1 above)
Comment (dpakach) [reply tests/unauthorized/userTest.js:18]: also the functionname request is a bit too generic, please change it to something like `addInvalidAuthInteraction`

**Review by individual-it — APPROVED** [review]

**Review by dpakach — COMMENTED** [review]
Was in review, but most of these things apply to many other files too so we can do it som other time

**Pull request merged**